checker/src/main/java/org/checkerframework/checker/determinism/DeterminismAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,990 @@
+        return types.isSubtype(types.erasure(tm), types.erasure(EnumerationTypeMirror));
+    }
+
+    /**
+     * Returns true if {@code receiverUnderlyingType} is Iterator and if the return type of {@code
+     * invokedMethodElement} is a type variable.
+     */
+    private boolean isIteratorNext(
+            TypeElement receiverUnderlyingType, ExecutableElement invokedMethodElement) {
+        if (isIterator(receiverUnderlyingType.asType())) {
+            if (invokedMethodElement.getSimpleName().contentEquals("next")
+                    && invokedMethodElement.getReturnType().getKind() == TypeKind.TYPEVAR
+                    && invokedMethodElement.getParameters().isEmpty()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns true if {@code receiverUnderlyingType} is AbstractList, AbstractSequentialList, or
+     * List and if the return type of {@code invokedMethodElement} is a type variable.
+     */
+    private boolean isAbstractListWithTypeVarReturn(
+            TypeElement receiverUnderlyingType, ExecutableElement invokedMethodElement) {
+        if (isAbstractList(receiverUnderlyingType.asType())
+                || isAbstractSequentialList(receiverUnderlyingType.asType())
+                || isList(receiverUnderlyingType.asType())) {
+            if (invokedMethodElement.getSimpleName().contentEquals("get")
+                    && invokedMethodElement.getReturnType().getKind() == TypeKind.TYPEVAR
+                    && invokedMethodElement.getParameters().size() == 1
+                    && invokedMethodElement.getParameters().get(0).asType().getKind()
+                            == TypeKind.INT) {
+                return true;
+            }
+            if (invokedMethodElement.getSimpleName().contentEquals("set")
+                    && invokedMethodElement.getReturnType().getKind() == TypeKind.TYPEVAR
+                    && invokedMethodElement.getParameters().size() == 2
+                    && invokedMethodElement.getParameters().get(0).asType().getKind()
+                            == TypeKind.INT
+                    && invokedMethodElement.getParameters().get(1).asType().getKind()
+                            == TypeKind.TYPEVAR) {
+                return true;
+            }
+            if (invokedMethodElement.getSimpleName().contentEquals("remove")
+                    && invokedMethodElement.getReturnType().getKind() == TypeKind.TYPEVAR
+                    && invokedMethodElement.getParameters().size() == 1
+                    && invokedMethodElement.getParameters().get(0).asType().getKind()
+                            == TypeKind.INT) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns true if {@code receiverUnderlyingType} is NavigableSet and if the return type of
+     * {@code invokedMethodElement} is a type variable.
+     */
+    private boolean isNavigableSetWithTypeVarReturn(
```

Comment:
Much of the code in this method can be replaced with `org.checkerframework.javacutil.ElementUtils#isMethod`.  And a few other methods in this class.